presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java

```diff
@@ -0,0 +1,246 @@
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.time.chrono.IsoChronology;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.Response.Status;
+
+/**
+ * Class managing retrieval and deletion of Presto logs
+ */
+public class LogsHandler
+{
+    // TODO: Make _all_ of these 'private static final' values configurable
+    private static final Path LOG_DIRECTORY = Paths.get("/tmp", "presto", "logs");
+    private static final Level LOG_ALL = Level.ALL;
+
+    private static final DateTimeFormatter DATE_FORMAT = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive().parseStrict()
+            .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+            .appendPattern("Z")
+            .toFormatter().withChronology(IsoChronology.INSTANCE);
+
+    private static final String DATE_GROUP = "date";
+    private static final String LEVEL_GROUP = "level";
+    private static final Pattern LOG_REGEX = Pattern.compile(
+            "^(?<" + DATE_GROUP + ">[0-9]{4}-[0-9]{2}-[0-9]{2}T" +
+                    "[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{3})?[+-][0-9]{4})" +
+                    "\t(?<" + LEVEL_GROUP + ">[A-Z]+)\t(?<thread>[^\t]+)" +
+                    "\t(?<class>[^\t]+)\t(?<message>.*)$");
+
+    private LogsHandler() {}
+
+    /**
+     * Method called in response to GET request
+     */
+    public static Response getLogs(String filename, JaxrsParameter<Instant> startDate,
+            JaxrsParameter<Instant> endDate, JaxrsParameter<Level> logLevel,
+            Integer maxEntries)
+    {
+        requireNonNull(startDate);
+        requireNonNull(endDate);
+        requireNonNull(logLevel);
+
+        if (!startDate.isValid() || !endDate.isValid()) {
+            return Response.status(Status.BAD_REQUEST)
+                    .entity("Invalid date format").build();
+        }
+        Instant start = startDate.get();
+        Instant end = endDate.get();
+
+        if (start != null && end != null && maxEntries != null) {
+            return Response.status(Status.BAD_REQUEST)
+                    .entity("Can not provide date range and limit number of entries")
+                    .build();
+        }
+
+        if (!logLevel.isValid()) {
+            return Response.status(Status.BAD_REQUEST)
+                    .entity("Invalid log level").build();
+        }
+        Level level = logLevel.get();
+
+        Path filePath = LOG_DIRECTORY.resolve(filename);
+        if (!Files.isRegularFile(filePath)) {
```

Comment:
A log file isn't necessarily named "something.log", so I haven't checked for that; if it's a file in the log directory and it has the name you gave, it can be read.